catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/ui/activity/AccessibilityProfilesActivityTest.java

```diff
@@ -0,0 +1,194 @@
+							parsedProfile = parser.getAttributeValue(null, AccessibilityPreferencesActivity.XML_NAME_FIELD);
+						}
+						if (parsedProfile.equals(profile)) {
+							parseAccessibilityProfileAttribute(parser, tagName);
+						}
+						break;
+					case XmlPullParser.TEXT:
+						break;
+					case XmlPullParser.END_TAG:
+						break;
+					default:
+						break;
+				}
+				event = parser.next();
+			}
+		} catch (XmlPullParserException ex) {
+			Log.e(TAG, Log.getStackTraceString(ex));
+		} catch (IOException ex) {
+			Log.e(TAG, Log.getStackTraceString(ex));
+		}
+	}
+
+	private void parseAccessibilityProfileAttribute(XmlPullParser parser, String tagName) {
+		if (!(tagName.equals(AccessibilityPreferencesActivity.XML_PARAMETER_TAG))) {
+			return;
+		}
+
+		SharedPreferences sharedPreferences = PreferenceManager
+				.getDefaultSharedPreferences(InstrumentationRegistry.getTargetContext());
+
+		boolean expectedValue = Boolean.parseBoolean(parser.getAttributeValue(null, AccessibilityPreferencesActivity.XML_VALUE_FIELD));
+		String attributeName = parser.getAttributeValue(null, AccessibilityPreferencesActivity.XML_NAME_FIELD);
+		if (attributeName.equals(ACCESS_LARGE_TEXT)) {
+			assertEquals(expectedValue, sharedPreferences.getBoolean(ACCESS_LARGE_TEXT, false));
+		} else if (attributeName.equals(ACCESS_HIGH_CONTRAST)) {
+			assertEquals(expectedValue, sharedPreferences.getBoolean(ACCESS_HIGH_CONTRAST, false));
+		} else if (attributeName.equals(ACCESS_ADDITIONAL_ICONS)) {
+			assertEquals(expectedValue, sharedPreferences.getBoolean(ACCESS_ADDITIONAL_ICONS, false));
+		} else if (attributeName.equals(ACCESS_LARGE_ICONS)) {
+			assertEquals(expectedValue, sharedPreferences.getBoolean(ACCESS_LARGE_ICONS, false));
+		} else if (attributeName.equals(ACCESS_HIGH_CONTRAST_ICONS)) {
+			assertEquals(expectedValue, sharedPreferences.getBoolean(ACCESS_HIGH_CONTRAST_ICONS, false));
+		} else if (attributeName.equals(ACCESS_LARGE_ELEMENT_SPACING)) {
+			assertEquals(expectedValue, sharedPreferences.getBoolean(ACCESS_LARGE_ELEMENT_SPACING, false));
+		} else if (attributeName.equals(ACCESS_STARTER_BRICKS)) {
+			assertEquals(expectedValue, sharedPreferences.getBoolean(ACCESS_STARTER_BRICKS, false));
+		} else if (attributeName.equals(ACCESS_DRAGNDROP_DELAY)) {
+			assertEquals(expectedValue, sharedPreferences.getBoolean(ACCESS_DRAGNDROP_DELAY, false));
+		} else if (attributeName.equals(ACCESS_FONTFACE)) {
+			String expectedFontface = parser.getAttributeValue(null, AccessibilityPreferencesActivity.XML_VALUE_FIELD);
+			if (expectedFontface.equals(ACCESS_FONTFACE_VALUE_STANDARD)) {
+				assertEquals(expectedFontface, ACCESS_FONTFACE_VALUE_STANDARD);
+			} else if (expectedFontface.equals(ACCESS_FONTFACE_VALUE_SERIF)) {
+				assertEquals(expectedFontface, ACCESS_FONTFACE_VALUE_SERIF);
+			} else if (expectedFontface.equals(ACCESS_FONTFACE_VALUE_DYSLEXIC)) {
+				assertEquals(expectedFontface, ACCESS_FONTFACE_VALUE_DYSLEXIC);
+			}
+		}
+	}
+}
```

Comment:
grundsätzlich find ich den test overkill.
eigentlich würd kein UI gebraucht werden, also am besten wäre hier stattdessen ein integrationTest ohne UI.

wenns denn dann umbedingt espresso sein muss, würd ich hergehen, und die
- testRule auf AccessibilityPreferenceActivity
- activity erst im setup (oder noch später, erst im test) mit intent launchen, wobei im intent das jeweilige testProfile (das es gar net im xml geben muss) mitübergeben
- im test schaun obs richtig befüllt wurde

Reply:
oder mit intending nachschaun dass die profile activity den richtigen intent schickt wenn ein profile geclickt wurde.

Reply:
I axed the xml testing, and checked the intents instead. So this now always stays in its own activiy.